node/clients/mdFactory.ts

```diff
@@ -90,4 +90,79 @@ export class MDFactory extends MasterData {
       fields: productReturnFields,
     })
   }
+
+  public updateSettingsSchema(schemaBody: typeof SETTINGS_SCHEMA) {
```

Comment:
@filafb @zekivtex 
Couldn't this be made only 1 method? So we later recursively check for schemas when accessing `checkAndUpdateSchemas()`

Reply:
You mean, one method that we pass the schema as an argument when calling it or a method like `createOrUpdateAllSchemas` and then we do the logic to update all schemas inside it?

Reply:
Yeah, I mean, there's only 1 dataEntity and a small enum of the schemas' names. Why we cannot have only 1 public method? And later in `checkAndUpdateSchemas` we can map all the schemas with only 1 function.

Reply:
Got it! We def could have only one public method. I was back and forth with that implementation, and tbh, I don't have a strong opinion which one is better.
  
The main reason I decided having different methods were: 
1. We are already creating a dictionary for the schemas inside the `mdFactory` client. Creating the enum, I saw it as a possibly two different source of truth for the schemas;
2. Having different methods allow us to be more strict with the schema type we are passing for every single method. The client manages the other arguments (data entity and schema name) for us;
3. In the next major I think we should go for a different data entity for every entity we have (`settings`, `requests`, `products`, `comments`, `history`). One method per entity mimics how it would be (one client per data entity). This structure would reduce the pressure on the indexation process. Per the [documentation](https://developers.vtex.com/vtex-rest-api/docs/master-data-schema-lifecycle#creating-a-schema), there is a background process validating all documents when creating a schema. The way the app is structured right now, when we update the schema for `products`, all documents, including the ones not related to products, will be considered. 

What do you think? Still consider having only one a better approach? Or should we start preparing for the `#3` I mentioned above and create different clients for different entities?

Reply:
Totally in with the `#3`! 😍